### PR TITLE
Allow other extensions to connect

### LINKED ISSUE #940: As a 3d party chrome app, I would like to communicate with Metamask
We're a [third party chrome app](https://fundrequest.io/) during its pre-development phase. One of our requirements would allow us to integrate the ethereum blockchain in an existing application (the **github frontend**, for now).

We want to avoid having to replicate the entire behaviour of metamask to perform transactions on the blockchain. Unfortunately, direct communication with Metamask is not possible due to everything we/you inject being sandboxed by the browser making sure no plugins can access other plugin's changes.

We've had a chat in the past:
https://twitter.com/QuintenDes/status/784373435522777088

Would it be possible to discuss this further here? What would the next steps be? 

Initial requirements would be: 
- call the "metamask" to open with prefilled values
- get a callback from metamask after such transaction has finished (done, failed, ..)

I'm willing to help with contributions should this story be approved.

### PR DESCRIPTION
**Pull request by jakubsta**
As discussed in #940 this PR allows to connect to MetaMask from other extensions.

Resolves #940

### CODE REVIEW

**Comment by danfinlay** [comment]
This looks good at a cursory glance, but please also understand we need to treat this carefully, and QA/security review it thoroughly before merging. I will try to prioritize completing this in the next sprint or the one after. Sorry for the wait, but I'm sure you don't want us neglecting security either ;)

**Comment by june07** [comment]
In the meantime here's a workaround...

https://june07.com/crypto/metamaskpay.htm?amount=1&payee=0x69F3C9210091A0E5cb7D01459683447173D4BDDA

It works for my extension [NiM](https://june07.com/nim)
MetaMask Send

**Comment by kumavis** [comment]
adding label to make sure this doesn't get in without extra review

**Comment by kumavis** [comment]
This does look good! Need to read up on the APIs and make sure they match our assumptions.

**Comment by kumavis** [comment]
Can you connect externally to `*`? I think I remember this being a problem when I looked at it a while back

**Comment by jakubsta** [comment]
Yes I can connect from external extension like here: https://github.com/jakubsta/example-extension/blob/master/background.js#L7 (it connects to extension built from this PR)

**Review by kumavis — APPROVED** [review]

**Comment by lazaridiscom** [comment]
@danfinlay / @kumavis, #940 is solved with e447438 merge?

**Pull request merged**
